### PR TITLE
refactor: decouple new router server configs from old management system

### PR DESCRIPTION
**Pull request by crepererum**
See commits. This would allow me to implement the router server on a more stable foundation.

Ref #2980.

### CODE REVIEW

File: generated_types/protos/influxdata/iox/router/v1/shard.proto [hunk 1]
```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
```
Comment (tustvold) [inline generated_types/protos/influxdata/iox/router/v1/shard.proto:1]: Is it intentional that this appears to be a duplicate instead of a move?

**Review by tustvold — APPROVED** [review]
This seems to just move the write buffer stuff into a write_buffer package and the routing stuff into a routing package with no major code changes. Could you maybe highlight any areas where that isn't the case if it isn't :sweat_smile:

**Comment by crepererum** [comment]
> This seems to just move the write buffer stuff into a write_buffer package and the routing stuff into a routing package with no major code changes. Could you maybe highlight any areas where that isn't the case if it isn't sweat_smile


For `shard.proto` it's not a move but a but a simplified copy.

**Comment by crepererum** [comment]
Protobuf is actually compatible:

1. The package move is OK because we don't put this information into `Any`.
2. The router-related changes are OK because this stuff is not used (yet).

**Pull request merged**